src/utils.js

```diff
@@ -56,27 +67,23 @@ export function getRootDirectory(file) {
   return path.normalize(dir).slice(root.length).split(path.sep)[0];
 }
 
-export function makeUrl(portalUrl, pathname, query = {}) {
-  const parsed = parse(portalUrl);
-
-  parsed.set("pathname", pathname);
-  parsed.set("query", query);
-  return parsed.toString();
-}
-
-export function makeUrlWithSkylink(portalUrl, endpointPath, skylink, query = {}) {
-  const parsedSkylink = parseSkylink(skylink);
-
-  return makeUrl(portalUrl, path.posix.join(endpointPath, parsedSkylink), query);
+/**
+ * Properly joins paths together to create a URL. Takes a variable number of
+ * arguments.
+ */
+export function makeUrl() {
+  let args = Array.from(arguments);
+  return args.reduce(function (acc, cur) {
+    return urljoin(acc, cur);
+  });
```

Comment:
I think this function doesn't make sense, it's doing exactly the same as `urljoin`.

`makeUrl('/a/', '/b/', '/c/')` === `urljoin('/a/', '/b/', '/c/')` ?

Reply:
`urljoin` failed the unit tests. I think if one of the arguments in the middle is `/` it will append it to a slash after it. I'll document this

Reply:
Then instead of arguments, we should use rest parameters. Usage of magic `arguments` is considered a bad practice due to it's magical nature :)

```suggestion
export function makeUrl(...args) {
  return args.reduce(function (acc, cur) {
    return urljoin(acc, cur);
  });
```